### PR TITLE
feature(input-data-model): add idm core with initial package use

### PR DESCRIPTION
**Pull request by mjreno**
* introduce dfn derived fortran input definition format
* add script to generate input definitions
* add generic input loader for mf6 file types
* add mem_set_value interface
* update dis, disv, disu, npf, dsp to source input from input data path

### CODE REVIEW

**Review by jdhughes-usgs — APPROVED** [review]
Left a few cursory comments that don't need to be addressed.

There is a powerpoint file checked in (`input_data_model.pptx')...should it be?

The IDM is more abstract than the previous way for getting data into the model. A developer document is needed to provide a roadmap for using it. I imaging it would not be too hard to look at the code to figure out how to do something but I think a developers document on using IDM to load the various blocks would be beneficial. Maybe with pointers on approaches for debugging data loaded in IDM and after IDM.

File: src/Model/GroundWaterFlow/gwf3dis8.f90 [hunk 1]
```diff
@@ -338,116 +331,100 @@ subroutine read_dimensions(this)
...
-    integer(I4B) :: n
-    integer(I4B) :: nvals
-    integer(I4B) :: ierr
-    logical :: isfound, endOfBlock
-    integer(I4B), parameter :: nname = 5
-    logical, dimension(nname) :: lname
-    character(len=24), dimension(nname) :: aname
-    character(len=300) :: ermsg
+    character(len=LENMEMPATH) :: idmMemoryPath
+    logical, dimension(5) :: afound
```
[the remark this answers is not in the document]
Comment (mjreno) [reply src/Model/GroundWaterFlow/gwf3dis8.f90:374]: I agree that this pattern is too mysterious and not ideal from a maintenance perspective.  What I was thinking at the time was a C-style enum which would assign variable names to indexes.  I'll give it some more thought.

File: src/Model/GroundWaterFlow/gwf3dis8idm.f90 [hunk 2]
```diff
@@ -0,0 +1,261 @@
+module GwfDisInputModule
```
[the remark this answers is not in the document]
Comment (mjreno) [reply src/Model/GroundWaterFlow/gwf3dis8idm.f90:1]: That is a good thought.  Another approach would be to add a github actions check to verify the input definitions are up to date when dfn's are changed.  I'm open to either but agree that we probably need a better way to ensure alignment.

File: src/Model/ModelUtilities/DiscretizationBase.f90 [hunk 3]
```diff
@@ -221,7 +221,7 @@ subroutine dis_ar(this, icelltype)
       end if
     end do
     !
-    if (this%writegrb) call this%write_grb(ict)
+    if (this%nogrb == 0) call this%write_grb(ict)
```
[the remark this answers is not in the document]
Comment (mjreno) [reply src/Model/ModelUtilities/DiscretizationBase.f90:224]: I agree it is counter-intuitive.  This was an attempt to align the user (input) side, NOGRB, with the internal variable.  I would have preferred changing the user variable but that probably wouldn't work due to historic compatibility, etc.  I could have left it but that would have created some other confusing flipping in the sourcing routine- I'll rethink as maybe there is still a cleaner way.

**Review by langevin-usgs — APPROVED** [review]

**Pull request merged**